core/src/main/scala/scalaz/StateT.scala

```diff
@@ -3,7 +3,7 @@ package scalaz
 import scala.annotation.tailrec
 import Id._
 
-sealed abstract class IndexedStateT[F[_], -S1, S2, A] { self =>
+sealed abstract class IndexedStateT[-S1, F[_], S2, A] { self =>
```

Comment:
`F[_]` should be the second to last parameter, i.e. `IndexedStateT[-S1, S2, F[_], A]`.

Reply:
Just pushed that but the build still fails. I'm assuming that the issue here is similar to EitherT in that type inference is not working correctly? `StateT` is ultimately a wrapper around `IndexedStateT`, so would there be a test that could prove/disprove this?

Reply:
If the problem is that the tests hang, it has nothing to do with type inference, since the code already compiled.